functions/eslint.config.mjs

```diff
@@ -0,0 +1,49 @@
+import js from '@eslint/js';
+import tseslint from 'typescript-eslint';
+import importPlugin from 'eslint-plugin-import';
+
+export default [
+  js.configs.recommended,
+  {
+    ignores: ['lib/**/*'],
+  },
+  // Base configuration for all files
+  {
+    languageOptions: {
+      globals: {
+        // Add Node.js globals
+        require: 'readonly',
+        process: 'readonly',
+        console: 'readonly',
+        __dirname: 'readonly',
+        __filename: 'readonly',
+        module: 'readonly',
+        exports: 'readonly'
+      }
+    }
+  },
+  // TS-specific configuration
+  {
+    files: ['**/*.ts'],
+    ...tseslint.configs.recommended[0],
+    languageOptions: {
+      parser: tseslint.parser,
+      parserOptions: {
+        project: ['./tsconfig.json'],
+        tsconfigRootDir: import.meta.dirname,
+      },
+    },
+    plugins: {
+      '@typescript-eslint': tseslint.plugin,
+      'import': importPlugin,
+    },
+  },
+  // Temporarily relax certain rules during migration
```

Comment:
I suggest to already create a ticket in linear or github issue to edit this after migration. Otherwise we risk to forget about it.